src-tauri/src/main.rs

```diff
@@ -210,7 +211,10 @@ async fn main() {
         })
         .on_window_event(move |event| {
             if let WindowEvent::CloseRequested { api, .. } = event.event() {
-                if wooting_macro_backend::config::ApplicationConfig::read_data().minimize_to_tray {
+                if ApplicationConfig::read_data()
+                    .expect("Error reading config")
+                    .minimize_to_tray
+                {
                     event.window().hide().unwrap();
                     api.prevent_close();
                 }
```

Comment:
You should consider what the class of errors are here and if it should be graceful handling or a panic like it will do now

Reply:
This function basically panics only if the config directory doesn't exist (not installed properly). This is to prevent a random .exe moving without proper installation of the application I think (due to the webview2 requirements and such)